scrapinghub/client.py

```diff
@@ -0,0 +1,350 @@
+import json
+
+from scrapinghub import APIError
+from scrapinghub import Connection
+from scrapinghub import HubstorageClient
+
+from .hubstorage.resourcetype import DownloadableResource
+from .hubstorage.resourcetype import ItemsResourceType
+
+# scrapinghub.hubstorage classes to use as-is
+from .hubstorage.activity import Activity
+from .hubstorage.frontier import Frontier
+from .hubstorage.job import JobMeta
+from .hubstorage.project import Reports
+from .hubstorage.project import Settings
+
+# scrapinghub.hubstorage proxied classes
+from .hubstorage.collectionsrt import Collections as _Collections
+from .hubstorage.collectionsrt import Collection as _Collection
+from .hubstorage.job import Items as _Items
+from .hubstorage.job import Logs as _Logs
+from .hubstorage.job import Samples as _Samples
+from .hubstorage.job import Requests as _Requests
+
+from .utils import DuplicateJobError, NonExistingSpider
+from .utils import LogLevel
+from .utils import get_tags_for_update
+from .utils import parse_project_id, parse_job_key
+from .utils import proxy_methods
+from .utils import wrap_kwargs
+
+
+class ScrapinghubClient(object):
+
+    def __init__(self, auth=None, dash_endpoint=None, **kwargs):
+        self.projects = Projects(self)
+        self._connection = Connection(apikey=auth, url=dash_endpoint)
+        self._hsclient = HubstorageClient(auth=auth, **kwargs)
+
+    def get_project(self, projectid):
+        return self.projects.get(parse_project_id(projectid))
+
+    def get_job(self, jobkey):
+        projectid = parse_job_key(jobkey).projectid
+        return self.projects.get(projectid).jobs.get(jobkey)
+
+    def close(self, timeout=None):
+        self._hsclient.close(timeout=timeout)
+
+
+class Projects(object):
+
+    def __init__(self, client):
+        self._client = client
+
+    def get(self, projectid):
+        return Project(self._client, parse_project_id(projectid))
+
+    def list(self):
+        return self._client._connection.project_ids()
+
+    def summary(self, **params):
+        return self._client._hsclient.projects.jobsummaries(**params)
+
+
+class Project(object):
+
+    def __init__(self, client, projectid):
+        self.id = projectid
+        self._client = client
+
+        # sub-resources
+        self.jobs = Jobs(client, projectid)
+        self.spiders = Spiders(client, projectid)
+
+        # proxied sub-resources
+        self.activity = Activity(client._hsclient, projectid)
```

Comment:
`proj.activity.list()` returns generator. I think we shouldn't leave non-proxied objects in the new client, it may cause confusion